src/python/gudhi/simplex_tree.pyx

```diff
@@ -806,13 +806,12 @@ cdef class SimplexTree:
         :returns: Serialized SimplexTree data structure
         :rtype: Byte Array
         """
-        cdef char* buffer
-        cdef size_t buffer_size = 0
-        with nogil:
-            buffer = self.get_ptr().serialize(buffer_size)
+        cdef size_t buffer_size = self.get_ptr().get_serialization_size()
+        # Let's use numpy to allocate a buffer. Will be deleted automatically
+        cdef char[:] buffer = np.empty(buffer_size, dtype='B')
+        self.get_ptr().serialize(&buffer[0], buffer_size)
         
-        return PyByteArray_FromStringAndSize(buffer, buffer_size)
-        del buffer
+        return PyByteArray_FromStringAndSize(&buffer[0], buffer_size)
```

Comment:
I think we might as well return the numpy array doesn't that work?

Reply:
Use numpy array, no more byte array for picklisation on [f8f3382](https://github.com/GUDHI/gudhi-devel/pull/810/commits/f8f3382e1d647ea56c6a72a2f74f3919a1e80eed).

Timings are even better:
## Rips de 250 points sur le 3-tore, expansion en dimension 3
| `vector::insert` | `num_simplices` | `numpy serialization` |
|---|---|---|
Rips creation: 0.002s            | Rips creation: 0.002s            | Rips creation: 0.002s
Simplex tree creation: 2.312s    | Simplex tree creation: 2.459s    | Simplex tree creation: 2.307s
Serialize Simplex_tree: 5.952s   | Serialize Simplex_tree: 3.588s   | Serialize Simplex_tree: 2.495s
Deserialize Simplex_tree: 4.380s | Deserialize Simplex_tree: 3.278s | Deserialize Simplex_tree: 2.472s
File size: 2.5Go

## Rips de 1000 points sur le 3-tore, expansion en dimension 2
| `vector::insert` | `num_simplices` | `numpy serialization` |
|---|---|---|
Rips creation: 0.033s            | Rips creation: 0.037s            | Rips creation: 0.032s
Simplex tree creation: 2.288s    | Simplex tree creation: 2.248s    | Simplex tree creation: 2.200s
Serialize Simplex_tree: 6.117s   | Serialize Simplex_tree: 3.582s   | Serialize Simplex_tree: 2.681s
Deserialize Simplex_tree: 4.559s | Deserialize Simplex_tree: 3.333s | Deserialize Simplex_tree: 2.492s
File size: 2.5Go

## Rips de 4000 points sur le 3-tore, expansion en dimension 1
| `vector::insert` | `num_simplices` | `numpy serialization` |
|---|---|---|
Rips creation: 0.317s            | Rips creation: 0.310s            | Rips creation: 0.305s
Simplex tree creation: 0.243s    | Simplex tree creation: 0.249s    | Simplex tree creation: 0.237s
Serialize Simplex_tree: 0.276s   | Serialize Simplex_tree: 0.181s   | Serialize Simplex_tree: 0.130s
Deserialize Simplex_tree: 0.251s | Deserialize Simplex_tree: 0.175s | Deserialize Simplex_tree: 0.116s
File size: 123Mo

## Rips de 8000 points sur le 3-tore, expansion en dimension 1
| `vector::insert` | `num_simplices` | `numpy serialization` |
|---|---|---|
Rips creation: 1.208s            | Rips creation: 1.216s            | Rips creation: 1.212s
Simplex tree creation: 1.003s    | Simplex tree creation: 1.077s    | Simplex tree creation: 1.009s
Serialize Simplex_tree: 1.083s   | Serialize Simplex_tree: 0.783s   | Serialize Simplex_tree: 0.512s
Deserialize Simplex_tree: 0.993s | Deserialize Simplex_tree: 0.686s | Deserialize Simplex_tree: 0.490s
File size: 489Mo

## Rips de 16000 points sur le 3-tore, expansion en dimension 1
| `vector::insert` | `num_simplices` | `numpy serialization` |
|---|---|---|
Rips creation: 4.871s            | Rips creation: 5.045s            | Rips creation: 4.729s
Simplex tree creation: 4.460s    | Simplex tree creation: 4.509s    | Simplex tree creation: 4.363s
Serialize Simplex_tree: 4.308s   | Serialize Simplex_tree: 2.862s   | Serialize Simplex_tree: 2.043s
Deserialize Simplex_tree: 4.024s | Deserialize Simplex_tree: 2.556s | Deserialize Simplex_tree: 1.835s
File size: 2Go